### PR TITLE
Create workflow that runs zizmor latest

### PR DESCRIPTION
**Pull request by colindean**
This is a ~copy of what's working for me in [keepachangelog-parser-java][1].

[1]: https://github.com/colindean/keepachangelog-parser-java/blob/master/.github/workflows/zizmor.yml

If accepted, this could probably be linked from README.md instead of the ESLint example link in #70.

### CODE REVIEW

**Comment by ncoghlan** [comment]
I reworked the PR in #70 to link to the location where this workflow will be in the repository.

I kept the ESLint link as well, though, since GitHub have some genuinely useful links in the preamble to their example.

File: .github/workflows/zizmor.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# https://github.com/woodruffw/zizmor
+name: GitHub Actions Security Analysis with Zizmor
+
+on:
+  push:
+    branches: ["master", "main"]
+  pull_request:
+    branches: ["**"]
```
Comment (woodruffw) [inline .github/workflows/zizmor.yaml:8]: Small thing: shouldn't this be `*`?

```suggestion
    branches: ["*"]
```

**Review by woodruffw — APPROVED** [review]
Thanks @colindean, this is great! One small comment (might be me not understanding)

File: .github/workflows/zizmor.yaml (hunk 1 above)
Comment (woodruffw) [reply .github/workflows/zizmor.yaml:8]: Ah nvm, TIL `**` means "match including `/`".

Comment (colindean) [reply .github/workflows/zizmor.yaml:8]: I'm not certain. I think that made its way into my example repo's config when I copy pasted somebody else's. Let's ship it as a fix and fix it later if it doesn't work I guess.

File: .github/workflows/zizmor.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# https://github.com/woodruffw/zizmor
+name: GitHub Actions Security Analysis with Zizmor
+
+on:
+  push:
+    branches: ["master", "main"]
```
Comment (woodruffw) [inline .github/workflows/zizmor.yaml:6]: Just for parsimony in this particular repo :slightly_smiling_face: 

```suggestion
    branches: ["main"]
```

**Comment by woodruffw** [comment]
Thanks @colindean!

**Pull request merged**